docs/requirements.txt

```diff
@@ -1,3 +1,3 @@
-sphinx==3.2.1
+sphinx==3.4
```

Comment:
Maybe, we can split this PR into multiple ones to simplify the review and also 
let's have sphinx version update in a separate PR.